### PR TITLE
ADAPT-2749: Dark Background Page

### PR DESCRIPTION
**Pull request by yvonnetangsu**
# READY FOR REVIEW

# Summary
- Note: I merged the "change basic card body text to rich text field" work into this PR to avoid the netlify build error due to schema change, so please review and merge that PR first before merging this one.
- Updated a few npm packages, including TW to v2.2.0
- Add the "Dark Background Page" which has a slightly different masthead, and dark background color with white text.
- This allows for building out the Perk landing page
- Mobile search and menu button now don't get squished at 200% zoom

Here's the Figma mock for perk landing page
Screen Shot 2021-06-18 at 12 30 36 AM

# Review By (Date)
- When does this need to be reviewed by?

# Criticality
- How critical is this PR on a 1-10 scale? Also see [Severity Assessment](https://stanfordits.atlassian.net/browse/D8CORE-1705).
- E.g., it affects one site, or every site and product?

# Review Tasks

## Setup tasks and/or behavior to test

1. https://deploy-preview-44--stanford-alumni.netlify.app/test/yvonne/test-dark-page/
2. Check that this page looks like the Figma mock Perks landing page
3. Check that the masthead now has a solid black background instead of a translucent black gradient.
4. https://deploy-preview-44--stanford-alumni.netlify.app/test/yvonne/test-dark-page-with-hero/
5. Check that when the dark page has a hero, the masthead uses the translucent black gradient.
6. Check that light theme pages are the same as before:
https://deploy-preview-44--stanford-alumni.netlify.app/test/yvonne/test-basic-page-interior-no-hero/
https://deploy-preview-44--stanford-alumni.netlify.app/test/yvonne/test-basic-page-interior-hero/
https://deploy-preview-44--stanford-alumni.netlify.app/test/yvonne/test-basic-page/
7. Look at masthead at XS breakpoint at 200% zoom, the search button and menu button shouldn't get distorted anymore.

## Front End Validation
- [ ] Is the markup using the appropriate semantic tags and passes HTML validation?
- [ ] Cross-browser testing has been performed?
- [ ] Automated accessibility scans performed?
- [ ] Manual accessibility tests performed?
- [ ] Design is approved by @ user?

### Code
- [ ] Are the naming conventions following our standards?
- [ ] Does the code have sufficient inline comments?
- [ ] Is there anything in this code that would be hidden or hard to discover through the UI?
- [ ] Are there any [code smells](https://blog.codinghorror.com/code-smells/)?
- [ ] Are tests provided? eg (unit, behat, or codeception)

### Code security
- [ ] Are all [forms properly sanitized](https://www.drupal.org/docs/8/security/drupal-8-sanitizing-output)?
- [ ] Any obvious [security flaws or new areas for attack](https://www.drupal.org/docs/8/security)?

## General
- [ ] Is there anything included in this PR that is not related to the problem it is trying to solve?
- [ ] Is the approach to the problem appropriate?

# Affected Projects or Products
- Does this PR impact any particular projects, products, or modules?

# Associated Issues and/or People
- JIRA ticket(s)
- Other PRs
- Any other contextual information that might be helpful (e.g., description of a bug that this PR fixes, new functionality that it adds, etc.)
- Anyone who should be notified? (`@mention` them here)

# Resources
- [AMP Tool](https://stanford.levelaccess.net/index.php)
- [Accessibility Manual Test Script](https://docs.google.com/document/d/1ZXJ9RIUNXsS674ow9j3qJ2g1OAkCjmqMXl0Gs8XHEPQ/edit?usp=sharing)
- [HTML Validator](https://validator.w3.org/)
- [Browserstack](https://live.browserstack.com/dashboard) and link to [Browserstack Credentials](https://asconfluence.stanford.edu/confluence/display/SWS/External+Account+Credentials)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -73,10 +73,10 @@
     "eslint-plugin-prettier": "^3.4.0",
     "eslint-plugin-promise": "^4.3.1",
     "eslint-plugin-react": "^7.23.2",
-    "postcss": "^8.2.15",
+    "postcss": "^8.3.5",
     "prettier": "^2.3.0",
     "react": "^16.14.0",
     "react-dom": "^16.14.0",
-    "tailwindcss": "^2.1.4"
+    "tailwindcss": "^2.2.0"
```
Comment (sherakama) [inline package.json:80]: There are a number of changes in this release that affect this PR. Please review https://github.com/tailwindlabs/tailwindcss/releases/tag/v2.2.0#user-content-changes-and-deprecations and make changes to this codebase to account for the difference in versions. 

eg: https://github.com/SU-SWS/saa_alumni/blob/ADAPT-2749-dark-page/src/styles/global.css#L15

Comment (sherakama) [reply package.json:80]: Please also review: https://github.com/tailwindlabs/tailwindcss/releases/tag/v2.2.0#user-content-transforms-and-filters-don-t-need-to-be-enabled-in-just-in-time-mode

> The transform, filter, and backdrop-filter classes aren't necessary for "enabling" those features when using the JIT engine:

We have a number of `su-transform` classes in the codebase.

Comment (yvonnetangsu) [reply package.json]: Looks like they're adding those back in for v2.2.2. I just changed out the @tailwind screens to variants part - thanks for the reminder.

Comment (yvonnetangsu) [reply package.json]: https://github.com/tailwindlabs/tailwindcss/releases/tag/v2.2.2

Comment (sherakama) [reply package.json]: https://github.com/tailwindlabs/headlessui/issues/628

> One quick fix is to just replace transform with relative on the foreground element. We're also going to push out v2.2.2 in a minute that will re-introduce the transform class just to create the stacking context even though it's no longer really necessary, just so people's existing stuff doesn't break in this way.

So, the re-introduction is to fix a breaking change problem. Please have a review if those classes are still necessary for our work.

Comment (yvonnetangsu) [reply package.json]: That's a good point. I'll review your Decanter one first, then deal with this one later today. Thanks!

File: src/components/composite/alert.js [hunk 2]
```diff
@@ -4,16 +4,22 @@ import { Alert as DecanterAlert, DismissButton } from "decanter-react";
...
-  const footerContent = (
-    <CreateBloks blokSection={alertCta} isLinkDark={isLinkDark} />
-  );
+  let footerContent = "";
+
+  if (hasCta) {
+    footerContent = (
+      <CreateBloks blokSection={alertCta} isLinkDark={isLinkDark} />
+    );
+  }
```
Comment (yvonnetangsu) [inline src/components/composite/alert.js:22]: @sherakama small alert fixup here for alert without a cta button

Comment (sherakama) [reply src/components/composite/alert.js:22]: Oh, thanks.

Comment (sherakama) [reply src/components/composite/alert.js:22]: Duplicated this work here: https://github.com/SU-SWS/ood_online_giving/pull/57

Comment (yvonnetangsu) [reply src/components/composite/alert.js:22]: Thanks!

**Comment by yvonnetangsu** [comment]
@sherakama It looks like we might be done here, since we're actually using `su-transform-gpu` instead of `su-transform` (I tested and it did improve performance and rendering). Looks like we would still need that class name.

**Review by sherakama — APPROVED** [review]
This looks great. Thanks for reviewing the TW update. The `isDark` prop gets drilled pretty deep and could probably benefit from a context but we can look at that at a later date. Nice work.

**Comment by yvonnetangsu** [comment]
> This looks great. Thanks for reviewing the TW update. The `isDark` prop gets drilled pretty deep and could probably benefit from a context but we can look at that at a later date. Nice work.

Yeah, totally agree. I was thinking a context approach for the theme would nice too. Thanks for the review.

**Pull request merged**